ledger-tool/src/main.rs

```diff
@@ -28,13 +36,189 @@ use std::{
+            if program_pubkey == solana_vote_program::id() { continue; }
+
+            let transaction_status: Option<RpcTransactionStatusMeta> = blockstore
+                .read_transaction_status((transaction.signatures[0], slot))
+                .unwrap_or_else(|err| {
+                    eprintln!(
+                        "Failed to read transaction status for {} at slot {}: {}",
+                        transaction.signatures[0], slot, err
+                    );
+                    None
+                })
+                .map(|transaction_status| transaction_status.into());
+
+            for instruction in &transaction.message.instructions {
+                let instruction_info = build_instruction_info(
+                    &instruction,
+                    &transaction,
+                );
+                instruction_wtr.serialize(&instruction_info);
+            }
+
+            let txn_info = build_txn_info(
+                slot,
+                &ledger_path,
+                &transaction,
+                &transaction_status
+            );
+            txn_wtr.serialize(&txn_info);
+        }
+    }
+    instruction_wtr.flush();
+    txn_wtr.flush();
+    Ok(())
+}
+
+fn build_instruction_info(
+    instruction: &CompiledInstruction,
+    transaction: &Transaction,
+) -> InstructionInfo {
+    let program_pubkey = transaction
+        .message
+        .account_keys[instruction.program_id_index as usize];
+
+    let mut instruction_info = InstructionInfo{
+        txn_sig: Some(transaction.signatures[0].to_string()),
+        program_pubkey: Some(bs58::encode(program_pubkey).into_string()),
+        ..Default::default()
+    };
+
+    if program_pubkey == solana_stake_program::id() {
+        if let Ok(stake_instruction) = limited_deserialize::<
+            solana_stake_program::stake_instruction::StakeInstruction,
+        >(&instruction.data)
+        {
+            instruction_info.program_instruction =
+                Some(format!("{:?}",stake_instruction));
+        }
+    } else if program_pubkey == solana_sdk::system_program::id() {
+        if let Ok(system_instruction) = limited_deserialize::<
+            solana_sdk::system_instruction::SystemInstruction,
+        >(&instruction.data)
+        {
+            instruction_info.program_instruction =
+                Some(format!("{:?}", system_instruction));
+        }
+    }
+
+
+    for account in &instruction.accounts {
+        instruction_info
+            .instruction_accounts
+            .push(bs58::encode(transaction
+                .message
+                .account_keys[*account as usize])
+                .into_string())
+    }
+    instruction_info
+}
+
+fn build_txn_info(
+    slot: Slot,
+    ledger_path: &PathBuf,
+    transaction: &Transaction,
+    transaction_status: &Option<RpcTransactionStatusMeta>
+) -> TransactionInfo {
+    let genesis_creation_time = open_genesis_config(&ledger_path).creation_time;
```

Comment:
Calling `open_genesis_config` on every `build_txn_info()` is pretty expensive, load it once and pass in a `&GenesisConfig`?